File: 0169-majority-element/0169-majority-element.py
class Solution: 
    def majorityElement(self, nums: List[int]) -> int: 
        d={} 
        n=len(nums) 
        for num in nums: 
            d[num]=d.get(num,0)+1 
        maj=n/2 
        for key,value in d.items(): 
            if value>maj: 
                maj=value
                a = key 
        return a

File: 0169-majority-element/0169-majority-element.c
int majorityElement(int* nums, int numsSize) {
    int candidate, count=0;
    for (int i=0;i<numsSize;i++){
        if (count==0) {
            candidate=nums[i];
            count++;
            continue;
        }
        else if (nums[i]==candidate) count++;
        else if (nums[i]!=candidate)count--;
        
    }
    return candidate;
}